packages/core/src/-private/milestone-coordinator.ts

```diff
@@ -63,28 +73,28 @@ export default class MilestoneCoordinator implements CoordinatorInterface {
   public deactivateAll(): void {
     this._continueAll();
 
-    this.names.forEach(name => {
-      delete ACTIVE_COORDINATORS[indexableKey(name)];
+    this.keys.forEach(key => {
+      delete ACTIVE_COORDINATORS[indexableKey(key)];
     });
 
-    this.names = [];
+    this.keys = [];
   }
 
   // Called from milestone()
-  public _milestoneReached<T extends PromiseLike<unknown>>(name: MilestoneKey, action: () => T): T {
+  public _milestoneReached<T extends PromiseLike<unknown>>(id: MilestoneKey, tags: MilestoneKey[], action: () => T): T {
     let target = this._nextTarget;
 
     // If we're already targeting another milestone, just pass through
-    if (target && target.name !== name) {
+    if (target && !matchesKey(target.key, id, tags)) {
       return action();
     }
 
     let deferred = defer();
-    if (target && target.name === name) {
-      this._targetReached(target, deferred, action);
+    if (target && matchesKey(target.key, id, tags)) {
+      this._targetReached(target, id, tags, deferred, action);
     } else {
-      assert(!this._pendingActions[indexableKey(name)], `Milestone '${name.toString()}' is already pending.`);
-      this._pendingActions[indexableKey(name)] = { deferred, action };
+      assert(!this._pendingActions[indexableKey(id)], `Milestone '${id.toString()}' is already pending.`);
+      this._pendingActions[indexableKey(id)] = { id: id, tags, deferred, action };
```

Comment:
There are many places where you call `this._pendingActions[indexableKey(id)]`. Are you avoiding using `this._getPendingAction(id)` because of the extra keys/id computation? 

It may be worth using that function or introducing another pending action getter for when you know the id you're passing in is the indexed key.

Reply:
What actually happened was those used to all be `this._pendingActions[name]`, which was simple enough that I never abstracted it. Then I find/replaced the `indexableKey()` wrapper in there when we added symbol support and didn't revisit that decision.

Just now I played with `_getPendingActionById(key)` and `_allPendingActions(): IterableIterator<PendingAction>` functions and ultimately it didn't feel like they helped much, because several of the points of access are the operand of a `delete` operator, so you can't abstract them behind a function call.